sdk/python/kfp/v2/components/utils.py

```diff
@@ -61,6 +61,12 @@ def maybe_rename_for_k8s(name: str) -> str:
                                     name.lower())).lstrip('-').rstrip('-')
 
 
+def sanitize_input_name(name: str) -> str:
+    """Sanitizes input name."""
+    return re.sub('-+', '-', re.sub('[^-_0-9A-Za-z]+', '-',
```

Comment:
Should we just reuse `maybe_rename_for_k8s`? you can keep the name `sanitize_input_name` if you prefer, but it can call the other method.

Reply:
Yeah, but the regex for `maybe_rename_for_k8s` and `sanitize_input_name` are different: 
`maybe_rename` doesn't allow underscore and cap letter, but `sanitize_input_name` allows it. 

Do you mean to use v1's function (add an additional parameter to `maybe_rename`) and produces two regex, one for non-underscore, one for underscore?

Reply:
Oh, I see. This needs to be different then. The two regex actually treat cap letters the same, because there is a `lower()` call before the replacement. But you're right on the underscore one. We can keep underscore here, but we have to convert dash to something else. Here's an example:
```
comp_op = load_component_from_text("""
name: my component
inputs:
  - {name: Input-1, type: String}
""")

comp_op(input_1='abc') # this is fine.
comp_op(input-1='abc') # this won't work because `input-1` is not a legal Python identifier.
```
How does v1 convert this? I'm guessing the regex should probably be something similar to
`re.sub('[^_0-9a-z]+', '_', name.lower())`

Reply:
Thanks, edited!